Upstream: Submitted - https://github.com/ponylang/ponyc/pull/1797
From 17d6d68f66c9056f3839fe143b0d98ba1d6dba34 Mon Sep 17 00:00:00 2001
From: BlackYoup <a.lefebvre@outlook.fr>
Date: Sat, 1 Apr 2017 23:47:55 +0200
Subject: [PATCH] Build: use AR if already defined and avoid the use of cc

Make the llvm include dir check also use $(CC) instead of cc.
Don't define AR if it's already defined.
---
 Makefile | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 8cbf30cb2..e011b6d46 100644
--- a/Makefile
+++ b/Makefile
@@ -9,8 +9,10 @@ else
   ifeq ($(UNAME_S),Linux)
     OSTYPE = linux
 
-    ifneq (,$(shell which gcc-ar 2> /dev/null))
-      AR = gcc-ar
+    ifndef AR
+      ifneq (,$(shell which gcc-ar 2> /dev/null))
+        AR = gcc-ar
+      endif
     endif
   endif
 
@@ -329,7 +331,7 @@ endif
 # (4) a list of the libraries to link against
 llvm.ldflags := $(shell $(LLVM_CONFIG) --ldflags)
 llvm.include.dir := $(shell $(LLVM_CONFIG) --includedir)
-include.paths := $(shell echo | cc -v -E - 2>&1)
+include.paths := $(shell echo | $(CC) -v -E - 2>&1)
 ifeq (,$(findstring $(llvm.include.dir),$(include.paths)))
 # LLVM include directory is not in the existing paths;
 # put it at the top of the system list
-- 
2.12.0
